packages/mobile/src/identity/feelessVerification.ts

```diff
@@ -0,0 +1,1061 @@
+    pepperQuotaRemaining = distributedBlindedPepper
+    requestQuotaRemaining = requestSubsidisedAttestation
+
+    // No pepper quota remaining is only bad if it's not already cached. Given Komenci will fetch
+    // a pepper for you once, a session could be invalid due to the pepper condition if a user
+    // fetched their pepper once this session then uninstalled without starting a new session
+    if (
+      (!ownPepper && !distributedBlindedPepper) ||
+      !requestSubsidisedAttestation ||
+      !submitMetaTransaction
+    ) {
+      sessionActive = false
+    }
+  }
+
+  yield put(
+    feelessUpdateVerificationState({
+      ...feelessVerificationState,
+      komenci: {
+        ...feelessVerificationState.komenci,
+        unverifiedMtwAddress,
+        sessionActive,
+        requestQuotaRemaining,
+        pepperQuotaRemaining,
+      },
+    })
+  )
+}
+
+function* fetchVerifiedMtw(contractKit: ContractKit, walletAddress: string, e164Number: string) {
+  Logger.debug(TAG, '@fetchVerifiedMtw', 'Starting fetch')
+  const feelessVerificationState: FeelessVerificationState = yield select(
+    feelessVerificationStateSelector
+  )
+
+  const phonHashDetails: PhoneNumberHashDetails = yield call(
+    fetchPhoneHashDetailsFromCache,
+    e164Number
+  )
+  const { phoneHash } = phonHashDetails
+  const lookupResult: IdentifierLookupResult = yield call(lookupAttestationIdentifiers, [phoneHash])
+  const possibleMtwAddresses = getAddressesFromLookupResult(lookupResult, phoneHash)
+
+  if (!possibleMtwAddresses) {
+    Logger.debug(TAG, '@fetchVerifiedMtw', 'No verified MTW found')
+    return null
+  }
+
+  // TODO: Incorporate fix for the error on this and remove try/catch block
+  let verificationResults: Array<Result<true, WalletValidationError>>
+  try {
+    verificationResults = yield all(
+      possibleMtwAddresses.map((possibleMtwAddress) =>
+        call(
+          verifyWallet,
+          contractKit,
+          possibleMtwAddress,
+          ALLOWED_MTW_IMPLEMENTATIONS,
+          walletAddress
+        )
+      )
+    )
+  } catch (error) {
+    Logger.debug(TAG, '@fetchVerifiedMtw', 'Unable to validate existing verified accounts')
+    Logger.error(TAG, '@fetchVerifiedMtw', error)
+    return null
+  }
+
+  const verifiedMtwAddresses = possibleMtwAddresses.filter(
+    (address, i) => verificationResults[i].ok
+  )
+
+  if (verifiedMtwAddresses.length > 2) {
+    throw Error(
+      'More than one verified MTW with walletAddress as signer found. Should never happen'
+    )
+  }
+
+  const verifiedMtwAddress = verifiedMtwAddresses[0]
+
+  if (!verifiedMtwAddress) {
+    Logger.debug(TAG, '@fetchVerifiedMtw', 'No verified MTW found')
+    return null
+  }
+
+  yield put(
+    feelessUpdateVerificationState({
+      ...feelessVerificationState,
+      status: {
+        isVerified: true,
+        numAttestationsRemaining: 0,
+        total: NUM_ATTESTATIONS_REQUIRED,
```

Comment:
Yeah that's a good call out. I will break this whole status object into a `VERIFIED_ATTESTATION_STATUS` constant with a note that it will need to change if our verification criteria ever does.

For now I don't think any change is needed because the `total` property is irrelevant for considerations of verification.